x-pack/plugins/features/server/plugin.ts

```diff
@@ -46,6 +46,14 @@ export interface PluginSetupContract {
    * */
   getElasticsearchFeatures(): ElasticsearchFeature[];
   getFeaturesUICapabilities(): UICapabilities;
+
+  /*
+   * In the future, OSS features should register their own subfeature
+   * privileges. This can be done when parts of Reporting are moved to
+   * src/plugins. For now, this method exists for `reporting` to tell
+   * `features` to include Reporting when registering OSS features.
+   */
+  enableReportingUiCapabilities(): void;
```

Comment:
cc @mshustov this implements the idea that came out of https://github.com/elastic/kibana/issues/94995. I hope that the above comment captures the tech debt comments that were brought up in that issue.